includes/SpecialRemovePII.php

```diff
@@ -0,0 +1,223 @@
+use Config;
+use ConfigFactory;
+use ExtensionRegistry;
+use FormSpecialPage;
+use GlobalRenameUser;
+use GlobalRenameUserDatabaseUpdates;
+use GlobalRenameUserLogger;
+use GlobalRenameUserStatus;
+use GlobalRenameUserValidator;
+use Html;
+use JobQueueGroup;
+use ManualLogEntry;
+use Status;
+use Title;
+use User;
+
+class SpecialRemovePII extends FormSpecialPage {
+	/** @var Config */
+	private $config;
+
+	/**
+	 * @param ConfigFactory $configFactory
+	 */
+	public function __construct( ConfigFactory $configFactory ) {
+		parent::__construct( 'RemovePII', 'handle-pii' );
+
+		$this->config = $configFactory->makeConfig( 'RemovePII' );
+	}
+
+	/**
+	 * @return array|string
+	 */
+	protected function getFormFields() {
+		if (
+			$this->config->get( 'RemovePIICentralWiki' ) &&
+			$this->config->get( 'DBname' ) !== $this->config->get( 'RemovePIICentralWiki' )
+		) {
+			return $this->msg( 'removepii-wiki-disabled' )->escaped();
+		}
+
+		$formDescriptor = [];
+		
+		$formDescriptor['warning'] = [
+			'type' => 'info',
+			'help' => Html::warningBox( $this->msg( 'removepii-warning-irreversible' )->parse() ),
+			'raw-help' => true,
+			'hide-if' => [ '!==', 'wpaction', 'removepii' ]
+		];
+
+		$formDescriptor['oldname'] = [
+			'type' => 'text',
+			'required' => true,
+			'label-message' => 'removepii-oldname-label'
+		];
+
+		$formDescriptor['newname'] = [
+			'type' => 'text',
+			'required' => true,
+			'label-message' => 'removepii-newname-label'
+		];
+		
+		$formDescriptor['action'] = [
+			'type' => 'select',
+			'options' => [
+				'Rename user' => 'renameuser',
+				'RemovePII' => 'removepii'
+			],
+			'required' => true,
+			'default' => 'renameuser',
+			'label-message' => 'removepii-action-label',
+			'help-message' => 'removepii-action-help'
+		];
+
+		return $formDescriptor;
+	}
+
+	/**
+	 * @param array $formData
+	 * @return Status
+	 */
+	public function validate( array $formData ) {
+		if ( !ExtensionRegistry::getInstance()->isLoaded( 'CentralAuth' ) ) {
+			return Status::newFatal( 'removepii-centralauth-notinstalled' );
+		}
+
+		if ( !ExtensionRegistry::getInstance()->isLoaded( 'Renameuser' ) ) {
+			return Status::newFatal( 'centralauth-rename-notinstalled' );
+		}
+
+		$oldUser = User::newFromName( $formData['oldname'] );
+		if ( !$oldUser ) {
+			return Status::newFatal( 'centralauth-rename-doesnotexist' );
+		}
+
+		if ( $oldUser->getName() === $this->getUser()->getName() ) {
+			return Status::newFatal( 'centralauth-rename-cannotself' );
+		}
+
+		$newUser = User::newFromName( $formData['newname'] );
+		if ( !$newUser ) {
+			return Status::newFatal( 'centralauth-rename-badusername' );
+		}
+
+		$validator = new GlobalRenameUserValidator();
+		$status = $validator->validate( $oldUser, $newUser );
+
+		return $status;
+	}
+
+	/**
+	 * @param array $formData
+	 * @return bool
+	 */
+	public function onSubmit( array $formData ) {
+		$out = $this->getOutput();
+
+		if ( $formData['action'] === 'renameuser' ) {
```

Comment:
Duplicating the global rename logic into this extension may be useful to reduce the need for a second manual action (renaming users), but is prone to errors. Do you think Miraheze can keep up with maintaining this piece of code?

Reply:
I think it can be maintained. It's only duplicates a small part of it and CentralAuth would only change something like that on 1.35->1.36 upgrades for example (major upgrades) so checking that before upgrading would make it easy to maintain. I did this because eventually I want to combine it into one step (the RemovePII  and Renameuser at the exact same time to prevent every logging the renames)

Reply:
Also just to add, with #15, this has an additional usage to never log renames done with removePII to never reveal the old name elsewhere.